### PR TITLE
Add Linear search and Queue

### PR DESCRIPTION
**Pull request by iamneha**

### CODE REVIEW

**Comment by iamneha** [comment]
@aktech I think it is ok now, can you please check this.
I am ready with more algorithm once you merge this then I will send more PR

**Comment by CuriousLearner** [comment]
@iamneha Just a quick suggestion, no need to wait till this PR is merged. You can work on different branches in git for submitting different PR's and when you get reviews, you just need to checkout on that particular branch and update.

Let me know if you need any help with it.

**Comment by iamneha** [comment]
@CuriousLearner thank you for your reply and I will send more PR asap.

File: pydsa/linear_search.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+def linear_search(List, item):
+    """
+    Find the element in list using linear search
+    >>> from pydsa import linear_search
+    >>> List = [3, 4, 6, 8, 12, 15, 26]
+    >>> item = 6
+    >>> linear_search(List, item)
+    2
+    """
+
```
Comment (aktech) [inline pydsa/linear_search.py]: pep8: W293 blank line contains whitespace

File: pydsa/__init__.py [hunk 2]
```diff
@@ -7,3 +7,5 @@
 from .sleep_sort import sleep_sort
 from .bfs import bfs
 from .binary_search import binary_search
+from .linear_search import linear_search
+from .queue import *
```
Comment (CuriousLearner) [inline pydsa/__init__.py]: You should not use `from module import *`

Please use here proper methods which you wish to import

**Comment by iamneha** [comment]
@CuriousLearner @aktech add all changes

File: pydsa/queue.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        self.List = []
+
+    def isEmpty(self):
+        return not len(self.List)
+
+    def enqueue(item):
+        """
+        Insert element in queue.
+        """
+        self.List.append(item)
```
Comment (aktech) [inline pydsa/queue.py]: `NameError: global name 'self' is not defined`

File: pydsa/tests/test_queue.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
+from pydsa.queue import Queue
+
+
+def test_queue():
+    def __init__self(self, size):
```
Comment (aktech) [inline pydsa/tests/test_queue.py]: This is not how the tests for the Queue should be written. You should first create a Queue object like this: `q = Queue()` and then do operations on it.

Comment (aktech) [inline pydsa/tests/test_queue.py]: No need to create function inside function here.

File: pydsa/tests/test_queue.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+        self.queue = Queue()
+        queue.size = 5
+    def test_enqueue(self):
+        self.queue.enqueue(3)
+        self.queue.enqueue(7)
+        self.queue.enqueue(13)
+        self.queue.enqueue(9)
+        self.assertEqual(size() == 3)
```
Comment (aktech) [inline pydsa/tests/test_queue.py]: Please see the documentation on how the tests are written in PyDSA.
https://github.com/pydsa/pydsa/wiki/Writing-and-Running-Tests-in-PyDSA

File: pydsa/tests/test_queue.py [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+    def test_dequeue(self):
+        self.dequeue()
+        self.assertTrue(size() == 2)
```
Comment (aktech) [inline pydsa/tests/test_queue.py]: https://github.com/pydsa/pydsa/pull/50/files#r59110898

**Comment by iamneha** [comment]
@aktech check please

**Comment by aktech** [comment]
It looks good to me. +1

**Pull request merged**